File: backend/app/schemas/growth.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, Field, field_serializer

from app.core.time import to_app_timezone


class GrowthProfileUpdate(BaseModel):
    session_id: str = Field(..., min_length=1, max_length=80)
    nickname: Optional[str] = Field(None, max_length=40)
    current_state: Optional[str] = Field(None, max_length=100)
    focus: Optional[str] = Field(None, max_length=80)
    personality: Optional[str] = Field(None, max_length=20)
    weekly_goal: Optional[str] = Field(None, max_length=200)
    setup_completed: Optional[bool] = None
    private_mode: Optional[bool] = None
    anonymous_default: Optional[bool] = None
    crisis_guard: Optional[bool] = None


class GrowthMemoryCreate(BaseModel):
    session_id: str = Field(..., min_length=1, max_length=80)
    category: str = Field("目标", min_length=1, max_length=40)
    content: str = Field(..., min_length=1, max_length=500)


class GrowthProfileItem(BaseModel):
    session_id: str
    nickname: Optional[str] = None
    current_state: Optional[str] = None
    focus: Optional[str] = None
    personality: str = "warm"
    weekly_goal: Optional[str] = None
    setup_completed: bool = False
    private_mode: bool = True
    anonymous_default: bool = False
    crisis_guard: bool = True
    created_at: datetime
    updated_at: datetime

    @field_serializer("created_at", "updated_at")
    def serialize_time(self, value: datetime) -> str:
        return to_app_timezone(value).isoformat()


class GrowthMemoryItem(BaseModel):
    id: int
    category: str
    content: str
    created_at: datetime

    @field_serializer("created_at")
    def serialize_created_at(self, value: datetime) -> str:
        return to_app_timezone(value).isoformat()


class GrowthStateItem(BaseModel):
    profile: GrowthProfileItem
    memories: list[GrowthMemoryItem]

File: frontend_update/frontend/src/api/growth.ts
import api from './client'

export interface GrowthProfileItem {
  session_id: string
  nickname?: string | null
  current_state?: string | null
  focus?: string | null
  personality: string
  weekly_goal?: string | null
  setup_completed: boolean
  private_mode: boolean
  anonymous_default: boolean
  crisis_guard: boolean
  created_at: string
  updated_at: string
}

export interface GrowthMemoryItem {
  id: number
  category: string
  content: string
  created_at: string
}

export interface GrowthStateItem {
  profile: GrowthProfileItem
  memories: GrowthMemoryItem[]
}

export interface GrowthProfileUpdate {
  session_id: string
  nickname?: string | null
  current_state?: string | null
  focus?: string | null
  personality?: string | null
  weekly_goal?: string | null
  setup_completed?: boolean
  private_mode?: boolean
  anonymous_default?: boolean
  crisis_guard?: boolean
}

export const getGrowthState = (sid: string) =>
  api.get<GrowthStateItem>(`/growth?session_id=${encodeURIComponent(sid)}`)

export const updateGrowthProfile = (payload: GrowthProfileUpdate) =>
  api.put<GrowthStateItem>('/growth', payload)

export const createGrowthMemory = (sid: string, category: string, content: string) =>
  api.post<GrowthMemoryItem>('/growth/memories', {
    session_id: sid,
    category,
    content,
  })

export const deleteGrowthMemory = (id: number, sid: string) =>
  api.delete(`/growth/memories/${id}?session_id=${encodeURIComponent(sid)}`)

export const clearGrowthState = (sid: string) =>
  api.delete(`/growth?session_id=${encodeURIComponent(sid)}`)
